File: qt/gen_qbasictimer.go
package qt

import (
	"unsafe"
)

type QBasicTimer struct {
	h          uintptr
	isSubclass bool
}

// NewQBasicTimer constructs a new QBasicTimer object.
func NewQBasicTimer() *QBasicTimer {
	ret := newQBasicTimer(QBasicTimer_new())
	ret.isSubclass = true
	return ret
}

func (this *QBasicTimer) Swap(other *QBasicTimer) {
	QBasicTimer_Swap(this.h, other.cPointer())
}

func (this *QBasicTimer) IsActive() bool {
	return (bool)(QBasicTimer_IsActive(this.h))
}

func (this *QBasicTimer) TimerId() int {
	return (int)(QBasicTimer_TimerId(this.h))
}

func (this *QBasicTimer) Id() TimerId {
	return (TimerId)(QBasicTimer_Id(this.h))
}

func (this *QBasicTimer) Start(msec int, obj *QObject) {
	QBasicTimer_Start(this.h, (int)(msec), obj.cPointer())
}

func (this *QBasicTimer) Start2(msec int, timerType TimerType, obj *QObject) {
	QBasicTimer_Start2(this.h, (int)(msec), (int)(timerType), obj.cPointer())
}

func (this *QBasicTimer) Start3(duration Duration, obj *QObject) {
	QBasicTimer_Start3(this.h, duration, obj.cPointer())
}

func (this *QBasicTimer) Start4(duration Duration, timerType TimerType, obj *QObject) {
	QBasicTimer_Start4(this.h, duration, (int)(timerType), obj.cPointer())
}

func (this *QBasicTimer) Stop() {
	QBasicTimer_Stop(this.h)
}

File: qt/gen_qbasictimer.cpp
// +build ignore

#include <QBasicTimer>
#include <QObject>
#include <qbasictimer.h>
#include "gen_qbasictimer.h"

#ifndef _Bool
#define _Bool bool
#endif

QBasicTimer* QBasicTimer_new() {
	return new QBasicTimer();
}

void QBasicTimer_Swap(QBasicTimer* self, QBasicTimer* other) {
	self->swap(*other);
}

bool QBasicTimer_IsActive(const QBasicTimer* self) {
	return self->isActive();
}

int QBasicTimer_TimerId(const QBasicTimer* self) {
	return self->timerId();
}

int QBasicTimer_Id(const QBasicTimer* self) {
	Qt::TimerId _ret = self->id();
	return static_cast<int>(_ret);
}

void QBasicTimer_Start(QBasicTimer* self, int msec, QObject* obj) {
	self->start(static_cast<int>(msec), obj);
}

void QBasicTimer_Start2(QBasicTimer* self, int msec, int timerType, QObject* obj) {
	self->start(static_cast<int>(msec), static_cast<Qt::TimerType>(timerType), obj);
}

void QBasicTimer_Start3(QBasicTimer* self, Duration duration, QObject* obj) {
	self->start(duration, obj);
}

void QBasicTimer_Start4(QBasicTimer* self, Duration duration, int timerType, QObject* obj) {
	self->start(duration, static_cast<Qt::TimerType>(timerType), obj);
}

void QBasicTimer_Stop(QBasicTimer* self) {
	self->stop();
}

void QBasicTimer_Delete(QBasicTimer* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QBasicTimer*>( self );
	} else {
		delete self;
	}
}

